### PR TITLE
Add fault injection in ZooKeeper client

### PR DESCRIPTION
**Pull request by tavplubix**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Detailed description / Documentation draft:
Randomly expire session in ZooKeeper client, enable fault injection in Stress tests

**Do not merge until some green commits appear in master.**

### CODE REVIEW

**Comment by tavplubix** [comment]
@Mergifyio update

**Comment by tavplubix** [comment]
@Mergifyio update

**Comment by tavplubix** [comment]
Wow, no failures except https://github.com/ClickHouse/ClickHouse/issues/39857
Let's try again

**Comment by tavplubix** [comment]
Stress test (memory) - https://github.com/ClickHouse/ClickHouse/pull/30826#issuecomment-1220610613

**Comment by tavplubix** [comment]
No failures in Stress Tests, let's try again

**Comment by tavplubix** [comment]
Integration tests (thread) [3/4] - `test_replicated_database/test.py::test_alter_detach_part` - https://github.com/ClickHouse/ClickHouse/pull/40875
Stress test (memory) - OOM in dmesg

So no related failures again

File: src/Common/ZooKeeper/ZooKeeperImpl.cpp [hunk 1]
```diff
@@ -633,13 +638,13 @@ void ZooKeeper::receiveThread()
...
     {
-        Int64 waited = 0;
+        Int64 waited_us = 0;
         while (!requests_queue.isFinished())
         {
             auto prev_bytes_received = in->count();
 
             clock::time_point now = clock::now();
-            UInt64 max_wait = operation_timeout.totalMicroseconds();
+            UInt64 max_wait_us = args.operation_timeout_ms;
```
Comment (antonio2368) [inline src/Common/ZooKeeper/ZooKeeperImpl.cpp:647]:
```suggestion
            UInt64 max_wait_us = args.operation_timeout_ms * 1000;
```

**Review by antonio2368 — COMMENTED** [review]
Cool idea

File: src/Common/ZooKeeper/ZooKeeperImpl.cpp (hunk 1 above)
Comment (tavplubix) [reply src/Common/ZooKeeper/ZooKeeperImpl.cpp:647]: Nice catch, thanks!

File: src/Common/ZooKeeper/ZooKeeperArgs.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+struct ZooKeeperArgs
+{
+    ZooKeeperArgs(const Poco::Util::AbstractConfiguration & config, const String & config_name);
+
+    /// hosts_string -- comma separated [secure://]host:port list
+    ZooKeeperArgs(const String & hosts_string);
+    ZooKeeperArgs() = default;
+    bool operator == (const ZooKeeperArgs &) const = default;
+    bool operator != (const ZooKeeperArgs &) const = default;
```
Comment (antonio2368) [inline src/Common/ZooKeeper/ZooKeeperArgs.h:25]: this should be generated with `operator==`

File: src/Common/ZooKeeper/ZooKeeperArgs.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#include <Common/ZooKeeper/ZooKeeperConstants.h>
+#include <Common/GetPriorityForLoadBalancing.h>
+
+namespace Poco
+{
+namespace Util
+{
+    class AbstractConfiguration;
+}
+}
```
Comment (antonio2368) [inline src/Common/ZooKeeper/ZooKeeperArgs.h:12]:
```suggestion
namespace Poco::Util
{
    class AbstractConfiguration;
}
```

**Review by antonio2368 — APPROVED** [review]

File: src/Common/ZooKeeper/ZooKeeperArgs.cpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        }
+        else
+            throw KeeperException(std::string("Unknown key ") + key + " in config file", Coordination::Error::ZBADARGUMENTS);
+    }
+
+    if (!chroot.empty())
+    {
+        if (chroot.front() != '/')
+            throw KeeperException(
+                std::string("Root path in config file should start with '/', but got ") + chroot, Coordination::Error::ZBADARGUMENTS);
```
[the remark this answers is not in the document]
Comment (antonio2368) [reply src/Common/ZooKeeper/ZooKeeperArgs.cpp]: Ah sorry, it's KeeperException and it doesn't support this form of arguments.

[the remark this answers is not in the document]
Comment (tavplubix) [reply src/Common/ZooKeeper/ZooKeeperArgs.cpp]: No problem, I added another ctor

**Comment by tavplubix** [comment]
Style Check — Check typos failed  :(

**Pull request merged**